src/components/Configurator.ts

```diff
@@ -1,42 +1,85 @@
-import path from 'path';
 import fs from 'fs';
 import yaml from 'js-yaml';
+import {mergeDeep} from './utils';
 
 export default class Configurator {
 
   path = '';
-  config = {} as any;
+  config: any = {};
   mapping = {
-    'S3_IMAGEBUCKETNAME': 'parameters.s3.imageBucketName',
+    'S3_BUCKETNAME': 'parameters.s3.bucketName',
     'S3_ENDPOINT': 'parameters.s3.endpoint',
+    'AWS_ACCESS_KEY_ID': 'parameters.aws.accessKeyId',
+    'AWS_SECRET_ACCESS_KEY': 'parameters.aws.secretAccessKey',
+    'ENV': 'parameters.env'
   };
 
 
   constructor (configPath = '/../config/config.yml') {
-    // this.config = yaml.load(fs.readFileSync(__dirname + configPath, 'utf8'));
-
+    this.config = yaml.load(fs.readFileSync(__dirname + configPath, 'utf8'));
+    this.selectAndApplyEnvParams();
   }
 
   selectAndApplyEnvParams () {
     // loop through mapping
-    // check if each key exists in process.env
-    // if yes, take this.config and replace value
-    // this.config should have updated values at the end
-
-    try {
-      console.log(this.config.parameters.s3);
-    } catch (err: any) {
-      console.error(err.stack);
-    }
+    const envConfig: any = {};
+    for (const [envVariable, configVariable] of Object.entries(this.mapping)) {
+      // check if each key exists in process.env
+      if (process.env[envVariable]) {
```

Comment:
how could they possibly not exist? You are looping through the mapping at line 9 so it has all of his props. Am I wrong?